{
   "assignee" : null,
   "body" : "From: https://travis-ci.org/bitcoin/bitcoin/jobs/133890912\r\n```\r\nsendheaders.py:\r\nInitializing test directory /tmp/testznw81v7u/9\r\nstart_node: bitcoind started, waiting for RPC to come up\r\nstart_node: RPC succesfully started\r\nstart_node: bitcoind started, waiting for RPC to come up\r\nstart_node: RPC succesfully started\r\nMiniNode: Connecting to Bitcoin Node IP # 127.0.0.1:11072\r\nMiniNode: Connecting to Bitcoin Node IP # 127.0.0.1:11072\r\nPart 1: headers don't start before sendheaders message...\r\nPart 1: success!\r\nPart 2: announce blocks with headers after sendheaders message...\r\nPart 2: success!\r\nPart 3: headers announcements can stop after large reorg, and resume after headers/inv from peer...\r\nUnexpected exception caught during testing: ConnectionRefusedError(111, 'Connection refused')\r\nStopping nodes\r\nWARN: Unable to stop node: CannotSendRequest('Request-sent',)\r\nNot cleaning up dir /tmp/testznw81v7u/9\r\nFailed\r\nstderr:\r\n bitcoind: main.cpp:2812: void PruneBlockIndexCandidates(): Assertion `!setBlockIndexCandidates.empty()' failed.\r\n  File \"/home/travis/build/bitcoin/bitcoin/bitcoin-x86_64-unknown-linux-gnu/qa/rpc-tests/test_framework/test_framework.py\", line 144, in main\r\n    self.run_test()\r\n  File \"/home/travis/build/bitcoin/bitcoin/bitcoin-x86_64-unknown-linux-gnu/qa/rpc-tests/sendheaders.py\", line 380, in run_test\r\n    new_block_hashes = self.mine_reorg(length=8)\r\n  File \"/home/travis/build/bitcoin/bitcoin/bitcoin-x86_64-unknown-linux-gnu/qa/rpc-tests/sendheaders.py\", line 240, in mine_reorg\r\n    self.nodes[1].invalidateblock(hash_to_invalidate)\r\n  File \"/home/travis/build/bitcoin/bitcoin/bitcoin-x86_64-unknown-linux-gnu/qa/rpc-tests/test_framework/coverage.py\", line 49, in __call__\r\n    return_val = self.auth_service_proxy_instance.__call__(*args, **kwargs)\r\n  File \"/home/travis/build/bitcoin/bitcoin/bitcoin-x86_64-unknown-linux-gnu/qa/rpc-tests/test_framework/authproxy.py\", line 137, in __call__\r\n    response = self._request('POST', self.__url.path, postdata)\r\n  File \"/home/travis/build/bitcoin/bitcoin/bitcoin-x86_64-unknown-linux-gnu/qa/rpc-tests/test_framework/authproxy.py\", line 123, in _request\r\n    self.__conn.request(method, path, postdata, headers)\r\n  File \"/usr/lib/python3.4/http/client.py\", line 1065, in request\r\n    self._send_request(method, url, body, headers)\r\n  File \"/usr/lib/python3.4/http/client.py\", line 1103, in _send_request\r\n    self.endheaders(body)\r\n  File \"/usr/lib/python3.4/http/client.py\", line 1061, in endheaders\r\n    self._send_output(message_body)\r\n  File \"/usr/lib/python3.4/http/client.py\", line 906, in _send_output\r\n    self.send(msg)\r\n  File \"/usr/lib/python3.4/http/client.py\", line 841, in send\r\n    self.connect()\r\n  File \"/usr/lib/python3.4/http/client.py\", line 819, in connect\r\n    self.timeout, self.source_address)\r\n  File \"/usr/lib/python3.4/socket.py\", line 509, in create_connection\r\n    raise err\r\n  File \"/usr/lib/python3.4/socket.py\", line 500, in create_connection\r\n    sock.connect(sa)\r\nPass: False, Duration: 17 s\r\n```\r\n\r\nThe root cause seems to be:\r\n```\r\n bitcoind: main.cpp:2812: void PruneBlockIndexCandidates(): Assertion `!setBlockIndexCandidates.empty()' failed.\r\n```",
   "closed_at" : "2016-05-30T13:37:15Z",
   "closed_by" : {
      "avatar_url" : "https://avatars.githubusercontent.com/u/126646?v=3",
      "events_url" : "https://api.github.com/users/laanwj/events{/privacy}",
      "followers_url" : "https://api.github.com/users/laanwj/followers",
      "following_url" : "https://api.github.com/users/laanwj/following{/other_user}",
      "gists_url" : "https://api.github.com/users/laanwj/gists{/gist_id}",
      "gravatar_id" : "",
      "html_url" : "https://github.com/laanwj",
      "id" : 126646,
      "login" : "laanwj",
      "organizations_url" : "https://api.github.com/users/laanwj/orgs",
      "received_events_url" : "https://api.github.com/users/laanwj/received_events",
      "repos_url" : "https://api.github.com/users/laanwj/repos",
      "site_admin" : false,
      "starred_url" : "https://api.github.com/users/laanwj/starred{/owner}{/repo}",
      "subscriptions_url" : "https://api.github.com/users/laanwj/subscriptions",
      "type" : "User",
      "url" : "https://api.github.com/users/laanwj"
   },
   "comments" : 2,
   "comments_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/8122/comments",
   "created_at" : "2016-05-30T13:22:32Z",
   "events_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/8122/events",
   "html_url" : "https://github.com/bitcoin/bitcoin/issues/8122",
   "id" : 157502030,
   "labels" : [
      {
         "color" : "d4c5f9",
         "name" : "Tests",
         "url" : "https://api.github.com/repos/bitcoin/bitcoin/labels/Tests"
      }
   ],
   "labels_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/8122/labels{/name}",
   "locked" : false,
   "milestone" : null,
   "number" : 8122,
   "repository_url" : "https://api.github.com/repos/bitcoin/bitcoin",
   "state" : "closed",
   "title" : "Intermittent travis faliure in sendheaders.py",
   "updated_at" : "2016-05-30T13:37:15Z",
   "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/8122",
   "user" : {
      "avatar_url" : "https://avatars.githubusercontent.com/u/126646?v=3",
      "events_url" : "https://api.github.com/users/laanwj/events{/privacy}",
      "followers_url" : "https://api.github.com/users/laanwj/followers",
      "following_url" : "https://api.github.com/users/laanwj/following{/other_user}",
      "gists_url" : "https://api.github.com/users/laanwj/gists{/gist_id}",
      "gravatar_id" : "",
      "html_url" : "https://github.com/laanwj",
      "id" : 126646,
      "login" : "laanwj",
      "organizations_url" : "https://api.github.com/users/laanwj/orgs",
      "received_events_url" : "https://api.github.com/users/laanwj/received_events",
      "repos_url" : "https://api.github.com/users/laanwj/repos",
      "site_admin" : false,
      "starred_url" : "https://api.github.com/users/laanwj/starred{/owner}{/repo}",
      "subscriptions_url" : "https://api.github.com/users/laanwj/subscriptions",
      "type" : "User",
      "url" : "https://api.github.com/users/laanwj"
   }
}
